pkg/discovery/discovery_client.go

```diff
@@ -0,0 +1,72 @@
+package discovery
+
+import (
+	"path"
+	"time"
+
+	"github.com/kcp-dev/apimachinery/pkg/logicalcluster"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes/scheme"
+	restclient "k8s.io/client-go/rest"
+)
+
+type ClusterDiscoveryClient interface {
+	Cluster(cluster logicalcluster.LogicalCluster) discovery.DiscoveryInterface
+}
+
+func NewClusterDiscoveryClientForConfig(c *restclient.Config) (ClusterDiscoveryClient, error) {
+
+	err := setDiscoveryDefaults(c)
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient, err := restclient.HTTPClientFor(c)
+	if err != nil {
+		return nil, err
+	}
+	client, err := restclient.UnversionedRESTClientForConfigAndClient(c, httpClient)
+	if err != nil {
+		return nil, err
+	}
+
+	return &clusterDiscoveryClient{config: c, restClient: client}, nil
+}
+
+type clusterDiscoveryClient struct {
+	restClient *restclient.RESTClient
+	config     *restclient.Config
+}
+
+func (c *clusterDiscoveryClient) Cluster(cluster logicalcluster.LogicalCluster) discovery.DiscoveryInterface {
+	scopedConfig := restclient.CopyConfig(c.config)
+	scopedConfig.Host = path.Join(scopedConfig.Host, cluster.Path())
```

Comment:
this does not parse as a URL, seems odd but definitely another sign that this is not how kubernetes wants me to do this